daffodil-core/src/test/scala/org/apache/daffodil/dsom/TestMiddleEndAttributes.scala

```diff
@@ -116,7 +115,8 @@ class TestMiddleEndAttributes {
   @Test def testStaticallyFirstWithChoice {
     val testSchema = SchemaUtils.dfdlTestSchema(
       <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>,
-      <dfdl:format representation="text" occursCountKind="parsed" lengthUnits="bytes" encoding="US-ASCII" initiator="" terminator="" separator="" ignoreCase="no"/>,
+      <dfdl:format representation="text" occursCountKind="parsed" lengthUnits="bytes" encoding="US-ASCII" initiator="" terminator="" separator="" ignoreCase="no"
+        alignment="1" alignmentUnits="bytes" leadingSkip="0" trailingSkip="0"/>,
```

Comment:
Well these properties are allowed to be required "from the get go" by the DFDL spec. Daffodil was not requiring them because the by-name parameters for the RunitmeData was allowing the error of not having them to be delayed until later, and at that point it got masked by some other error. 

When I removed the by-name from this, now the test fails earlier with lack of alignment and leading/strailing skip, rather than getting the expected error. Putting in these properties enables the test to continue to test what it was intended to illustrate.